crates/bevy_picking_core/src/focus.rs

```diff
@@ -0,0 +1,133 @@
+use crate::{PausedForBlockers, PickableTarget};
+use bevy::{prelude::*, ui::FocusPolicy};
+
+/// Tracks the current hover state to be used with change tracking in the events system.
+///
+/// # Requirements
+///
+/// An entity with the `Hover` component must also have an [Interaction] component.
+#[derive(Component, Debug, Default, Copy, Clone)]
+pub struct Hover {
+    hovered: bool,
+}
+
+impl Hover {
+    pub fn hovered(&self) -> bool {
+        self.hovered
+    }
+}
+
+/// Marker component for entities that, whenever their [Interaction] component is anything other
+/// than `None`, will suspend highlighting and selecting [PickableTarget]s. Bevy UI [Node]s have this
+/// behavior by default.
+#[derive(Component)]
+pub struct PickingBlocker;
```

Comment:
This seems extremely nice; I'd like this upstreamed.

Reply:
Wow same! Currently using this for a level editor coupled with `bevy_egui` and... Yup I could use this 😄